app/models/organization.rb

```diff
@@ -9,14 +13,54 @@ class Organization < ActiveRecord::Base
 
   before_save :add_county
 
-  def primary_address
-    addresses.first
+  def self.find_any(id)
+    unscoped.find(id)
+  end
+
+  def self.find_deleted(id)
+    deleted.find id
+  end
+
+  def self.deleted
+    unscoped.where.not(deleted_at: nil)
+  end
+
+  def self.not_deleted
+    where(deleted_at: nil)
+  end
+
+  def soft_delete
+    return false if open_orders.present?
+
+    transaction do
+      organization_users.map(&:destroy!) if organization_users.present?
+
+      self.deleted_at = Time.zone.now
+      save
+    end
+  end
+
+  def restore
+    self.deleted_at = nil
+    save!
   end
 
   def self.counties
     pluck(:county).uniq
   end
 
+  def deleted?
+    deleted_at != nil
+  end
+
+  def primary_address
+    addresses.first
+  end
+
+  def open_orders
+    @_open_orders ||= orders.select(&:open?)
+  end
```

Comment:
I think this could become an association, something like: `has_many :open_orders, class_name: "Order", -> { where(status: Order.open_statuses) }` (I would then add `open_statuses` to the OrderStatus concern, like similar methods already there... also if you prefer it could be `where.not(status: Order.closed_statuses)`). See https://stackoverflow.com/a/38484808/122 for more details.

Reply:
👍